python/cudf_polars/cudf_polars/experimental/scheduler.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+"""Synchronous task scheduler."""
+
+from __future__ import annotations
+
+from collections.abc import MutableMapping
+from typing import TYPE_CHECKING, Any, TypeVar, Union
+
+if TYPE_CHECKING:
+    from collections.abc import Mapping
+    from typing import TypeAlias
+
+
+Key: TypeAlias = str | tuple[Union["Key", int], ...]
```

Comment:
Just making sure I understand this type: what are our allowed Keys?

1. `str` like `'groupby-<hash>'`
2. Tuple like `('groupby-<hash>', 0)`

Do we need to accept both of those? Or can we use just the tuple form?

And is the second member of the `Union` correct? That it's either a **Key** or int? Should this maybe be

```
Key: TypeAlias = str | tuple[str, int]
```

(if we need both forms, and if I understood what's allowed here)

Reply:
>Do we need to accept both of those? Or can we use just the tuple form?

Our **final** key is always a `str` for now.

>Should this maybe be...

Yeah, I was thinking we could have keys like `tuple[str, int, int]`, but we could certainly be more strict.

Reply:
Using `Key: TypeAlias = str | tuple[str, int]` for now.